### PR TITLE
Improve Routes

### PR DESCRIPTION
**Pull request by aayush-05**
Fix unnnecessary routes and improve code layout for some

### CODE REVIEW

**Comment by aayush-05** [comment]
@tachyons

File: config/routes.rb [hunk 1]
```diff
@@ -15,10 +15,10 @@
   # resources :assignment_submissions
   resources :group_members, only: %i[create destroy]
   resources :groups, except: %i[index] do
-    resources :assignments
+    resources :assignments, except: %i[index]
```
Comment (tachyons) [inline config/routes.rb:18]: Was this route in use before ?

Comment (aayush-05) [reply config/routes.rb:18]: No, afaik

**Comment by tachyons** [comment]
specs are failing, rebase with latest master

**Comment by aayush-05** [comment]
Done

**Pull request merged**